packages/aws-cdk/lib/version.ts

```diff
@@ -24,13 +24,8 @@ function commit(): string {
 
 export class VersionCheckTTL {
   public static timestampFilePath(): string {
-    // Get the home directory from the OS, first. Fallback to $HOME.
-    const homedir = os.userInfo().homedir || os.homedir();
-    if (!homedir || !homedir.trim()) {
-      throw new Error('Cannot determine home directory');
-    }
     // Using the same path from account-cache.ts
-    return path.join(homedir, '.cdk', 'cache', 'repo-version-ttl');
+    return path.join(getCdkHome(), 'cache', 'repo-version-ttl');
```

Comment:
this should be `path.join(getCdkHome(), '.cdk', 'cache', ...)`

Reply:
This would result in `path.join(process.env.CDK_HOME, '.cdk', 'cache')` if set, instead of the current `path.join(process.env.CDK_HOME, 'cache')`

Reply:
Seems odd.. It means that CDK_HOME has two different meanings. In one instance it refers the `~` (for the sake of `~/.cdk.json`) and in the other it refers to `~/.cdk` (for the sake of the cache directory).

Maybe ew should just add `CDK_CACHE_DIR` to resolve the cache location issue and leave the rest as-is.

Reply:
If we're going in that direction, maybe CDK_CACHE_DIR should be defaulted to a directory in `os.tmpdir()` instead of `~/.cdk/`?

Reply:
Sounds good to me